src/attrib.d

```diff
@@ -411,6 +411,11 @@ public:
     {
         assert(msg);
         char* depmsg = null;
+        sc = sc.startCTFE();
+        this.msg = this.msg.semantic(sc);
+        this.msg = resolveProperties(sc, this.msg);
+        sc = sc.endCTFE();
+        this.msg = this.msg.optimize(WANTexpand);
```

Comment:
Thanks for the tip.
So it either have to be part of semantic (X) or `scope` / `dsymbol` will need to be changed store the `DeprecatedDeclaration` instead of just the message (as it's done with UDA currently) ? I guess the later would be cleaner (albeit more invasive) ?